File: app-goconecta/app-goconecta.Server/DTOs/HotelDTO.cs
using app_goconecta.Server.Models;
using System.Collections.Generic;
using System.Linq;

namespace app_goconecta.Server.DTOs;

public class HotelDTO
{
    public int Id { get; set; }
    
    public string? Name { get; set; }
    public string? Description { get; set; }
    public int Rating { get; set; }
    
    public string? Region { get; set; }
    public string? Address { get; set; }
    public List<string> Amenities { get; set; }

    public static HotelDTO FromModel(Hotel hotel)
    {
        return new HotelDTO
        {
            Id = hotel.Id,
            Name = hotel.Name,
            Description = hotel.Description,
            Rating = hotel.Rating,
            Region = hotel.Region,
            Address = hotel.Address,
            Amenities = hotel.Amenities?.Select(a => a.Name).ToList() ?? new List<string>()
        };
    }
    
}

File: app-goconecta/app-goconecta.client/src/dtos/HotelDTO.js
// DTO para hotel
export class HotelDTO {
  /**
   * @param {Object} params
   * @param {string|number} params.id
   * @param {string} params.name
   * @param {string} params.description
   * @param {number} params.rating
   * @param {string} params.region
   * @param {string} params.address
   */
  constructor({ id, name, description, rating, region, address }) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.rating = rating;
    this.region = region;
    this.address = address;
  }
}
